lib/charms/layer/jenkins/plugins.py

```diff
@@ -39,16 +39,18 @@ def install(self, plugins):
         plugins = plugins or ""
         plugins = plugins.split()
         plugins = self._get_plugins_to_install(plugins)
-
+        configured_plugins = hookenv.config()["plugins"].split()
         host.mkdir(
             paths.PLUGINS, owner="jenkins", group="jenkins", perms=0o0755)
-        existing_plugins = set(glob.glob("%s/*.jpi" % paths.PLUGINS))
+        existing_plugins = set(glob.glob("%s/*.?pi" % paths.PLUGINS))
         try:
-            installed_plugins = self._install_plugins(plugins)
+            self._install_plugins(plugins)
         except Exception:
             hookenv.log("Plugin installation failed, check logs for details")
             raise
 
+        plugin_file_names = tuple(map(lambda x: "/{}.jpi".format(x), configured_plugins))
```

Comment:
How does this interact with the hpi/jpi matching above?

Reply:
This is only used the get the file names of the configured plugins from their names, e.g: ace-editor -> ace-editor.jpi. Then on the line below it will use those files to find all plugins that need to be kept(`installed_plugins`) and finally it'll get all the plugin files in `existing_plugins` minus the plugins to keep.